js/sliderView.js

```diff
@@ -289,74 +291,77 @@ define([
       } else if (this.$('.slider__widget').hasClass('show-correct-answer')) {
         this.showCorrectAnswer();
       }
-    },
+    }
 
-    showCorrectAnswer: function() {
-      var answers = [];
+    showCorrectAnswer() {
+      const answers = [];
+      this.showScaleMarker(false);
 
+      //are we dealing with a single correct answer or a range?
       if (this.model.has('_correctAnswer')) {
-        var correctAnswer = this.model.get('_correctAnswer');
-      }
+        const correctAnswer = this.model.get('_correctAnswer');
 
-      if (this.model.has('_correctRange')) {
-        var bottom = this.model.get('_correctRange')._bottom;
-        var top = this.model.get('_correctRange')._top;
-        var step = (this.model.has('_scaleStep') ? this.model.get('_scaleStep') : 1);
+        if (correctAnswer) {
+          answers.push(correctAnswer);
+        }
       }
 
-      this.showScaleMarker(false);
-
-      //are we dealing with a single correct answer or a range?
-      if (correctAnswer) {
-        answers.push(correctAnswer);
-      } else if (bottom !== undefined && top !== undefined) {
-        var answer = this.model.get('_correctRange')._bottom;
-        var topOfRange = this.model.get('_correctRange')._top;
-        while (answer <= topOfRange) {
-          answers.push(answer);
-          answer += step;
+      if (answers.length === 0) {
```

Comment:
Use variables to increase clarity and reduce nesting. Apply "return early" principle.
```js
      const correctRange = this.model.has('_correctRange');
      const hasCorrectRange = (correctRange !== undefined);

      if (!hasCorrectAnswer && !hasCorrectRange) {
        console.log('adapt-contrib-slider::WARNING: no correct answer or correct range set in JSON');
      }
      
      if (!hasCorrectAnswer && hasCorrectRange) {
        const bottom = correctRange ._bottom;
        const top =correctRange ._top;
        const step = (this.model.has('_scaleStep') ? this.model.get('_scaleStep') : 1);

        if (bottom !== undefined && top !== undefined) {
          const answer = this.model.get('_correctRange')._bottom;
          const topOfRange = this.model.get('_correctRange')._top;
          while (answer <= topOfRange) {
            answers.push(answer);
            answer += step;
          }
        }
      }
```

Reply:
all great suggestions thanks, addressing them currently.

would you oppose if I was to make `if (!hasCorrectAnswer && hasCorrectRange) {`
`else if (hasCorrectRange) {` ?

Reply:
Nope I wouldn't oppose, that seems fine.

Reply:
Thanks @oliverfoster for the suggestion – the more I read the code, the more I feel there's enough going on to warrant a separate function. Something like:
```js
getCorrectAnswers() {
  const answerSingle = this.model.get('_correctAnswer');
  if (answerSingle) {
    return [ answerSingle ];
  }
  const answerMultiple = this.model.get('_correctRange');
  const answers = [];
  if (!answerMultiple) {
    console.log('adapt-contrib-slider::WARNING: no correct answer or correct range set in JSON');
    return answers;
  }
  const bottom = answerMultiple._bottom;
  const top = answerMultiple._top;
  if (bottom === undefined || top === undefined) {
    return answers;
  }
  let answer = bottom;
  const step = this.model.get('_scaleStep') || 1;
  while (answer <= top) {
    answers.push(answer);
    answer += step;
  }
  return answers;
}
```

Reply:
Yea, that block in the middle does seem somewhat out of place.

Reply:
I like this idea - splits the code up a bit meaning you have a show and a get function 👍 

I was also thinking that we should only attempt to animate to position/show model answer and set slider value when there are answers. This means we could return after logging out the warning when there are no answers.

For example, 

`const answers = this.getCorrectAnswers();
  if (!answers.length) {
    console.log('adapt-contrib-slider::WARNING: no correct answer or correct range set in JSON');
    return
  } else {
    const middleAnswer = answers[Math.floor(answers.length / 2)];
    this.animateToPosition(this.mapIndexToPixels(this.getIndexFromValue(middleAnswer)));
    this.showModelAnswers(answers);
    this.setSliderValue(middleAnswer);
  }
` 

Let me know your thoughts!